core/geometry/src/curve/ProxyCurve.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module Curve
+ */
+
+import { CurvePrimitive } from "../curve/CurvePrimitive";
+import { GeometryHandler, IStrokeHandler } from "../geometry3d/GeometryHandler";
+import { Plane3dByOriginAndUnitNormal } from "../geometry3d/Plane3dByOriginAndUnitNormal";
+import { Plane3dByOriginAndVectors } from "../geometry3d/Plane3dByOriginAndVectors";
+import { Point3d } from "../geometry3d/Point3dVector3d";
+import { Range3d } from "../geometry3d/Range";
+import { Ray3d } from "../geometry3d/Ray3d";
+import { Transform } from "../geometry3d/Transform";
+import { LineString3d } from "./LineString3d";
+import { StrokeOptions } from "./StrokeOptions";
+
+/**
+ * A ProxyCurve is expected to be used as a base class for
+ * curve types that use some existing curve (the proxy) for evaluation and display
+ * but carry other defining data.
+ * * The ProxyCurve implements all required CurvePrimitive methods by dispatching
+ *    to the proxy.
+ * * These methods presumably require support from the application class and are left abstract:
+ *    * clone
+ *    * clonePartialCurve
+ *    * cloneTransformed
+ *    * curvePrimitiveType

```

Comment:
Based on the documentation for `CurvePrimitiveType`, people write code like:
`switch (curve.curvePrimitiveType) {
  case CurvePrimitiveType.Arc:
    (curve as Arc3d).doArc3dStuff();
`

Such code will break if `curve` is a ProxyCurve.

Reply:
This extensibility issue has always existed, both for the string and for the double dispatch handleXXXX methods.  I'm mildly surprised it hasn't caused more problems.

InterpolationCurve triggered the issue in backend geometrySummary.ts -- incomplete case statement.
That particular one was fixed in this commit.

What would you think of
1) replace the the geometrySummary.ts case statement with a call to
      description  = IModelJson.Writer.toIModelJson(geometry);
2) for the low-verbosity executions, possibly strip out arrays from the description.

(I don't think ProxyCurve itself should never be a final class.)

Reply:
That doesn't address the common problem of "I have a CurvePrimitive, I want to do subclass-specific stuff with it."
Currently there is exactly one subclass of ProxyCurve, right? Do you intend to add additional ones? If not, why bother with the intermediate class?

Reply:
ProxyCurve is still abstract (due to inheritance from GeometryQuery and Curveprimitive) so there will be no instances.

It will probably be the base class for additional types -- e.g. Akima curve if we get forced to support it.

Reply:
Ok. Presumably AkimaCurve would introduce a new CurvePrimitiveType "akimaCurve", right? My concern was somebody writing a ProxyCurve that, e.g., is not an Arc3d but implements `curvePrimitiveType` to return "arc". As long as that doesn't happen there's not a problem.